### PR TITLE
TW-2036: Update readme with customized homeserver

### PR DESCRIPTION
**Pull request by nqhhdev**
## Ticket
  — #2036 

### Noted:

  — Need to merge the PR first #2067 2067

## Use docker desktop

https://github.com/user-attachments/assets/a5aebd09-99c6-450e-a1b0-5a92d6599541

## user docker CLI

https://github.com/user-attachments/assets/64614ff1-d570-4daa-aa97-b22a747199f5

https://github.com/user-attachments/assets/7be5480b-79e3-499d-b3a0-bac5fa6d6def

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +77,6 @@ bash scripts/config-pre-commit.sh
 ```
 
 # Special thanks
-* <a href="https://github.com/krille-chan/fluffychat">FluffyChat</a> is the original repository of this project. A huge thanks to the upstream repository for their vital contributions, not only for this project but also for [Matrix SDK in Dart](https://github.com/famedly/matrix-dart-sdk)
```
Comment (hoangdat) [inline README.md:80]: should not delete it

File: README.md [hunk 2]
```diff
@@ -92,3 +91,35 @@ bash scripts/config-pre-commit.sh
 * <a href="https://github.com/madsrh/WoodenBeaver">WoodenBeaver</a> sound theme for the notification sound.
 
 * The Matrix Foundation for making and maintaining the [emoji translations](https://github.com/matrix-org/matrix-doc/blob/main/data-definitions/sas-emoji.json) used for emoji verification, licensed Apache 2.0
+
+### Setup to build web version using Docker
+
+- Go to the [docker hub](https://hub.docker.com/r/linagora/twake-web/tags)
+- Pull the image using `docker pull linagora/twake-web-builder:<tag>`
```
Comment (hoangdat) [inline README.md:98]:
```suggestion
- Pull the image using `docker pull linagora/twake-web:<tag>`
```

File: README.md [hunk 3]
```diff
@@ -92,3 +91,35 @@ bash scripts/config-pre-commit.sh
...
+            - Config the container name
+            - Config the host port
+            - Config the volume with host path and container path
+            - Host path should be the root of the the config.json file (.../config.json)
+            - Container path should be the root of the project: /usr/share/nginx/html/web/config.json
+            - Click on Run
+            - Open the browser and go to `http://localhost:<hostport>/web/`
+    - Run the image using docker CLI
+        - Open terminal and login to docker
+            - `docker login`
```
Comment (hoangdat) [inline README.md:115]: no need

File: README.md [hunk 4]
```diff
@@ -92,3 +91,35 @@ bash scripts/config-pre-commit.sh
...
+        - Check the image you pulled
+            - `docker images`
```
Comment (hoangdat) [inline README.md:117]: no need

File: README.md [hunk 5]
```diff
@@ -92,3 +91,35 @@ bash scripts/config-pre-commit.sh
...
+        - Pull the image using the following command:
+            - `docker pull linagora/twake-web:<tag>`
+        - Verify the image of twake-web is pulled
+            - `docker images`
```
Comment (hoangdat) [inline README.md:121]: no need

File: README.md [hunk 6]
```diff
@@ -92,3 +91,35 @@ bash scripts/config-pre-commit.sh
...
+        - Run the image using the following command:
+            - `docker run -d -p <host port>:80 --name <container name> -v <host path>:/usr/share/nginx/html/web/config.json linagora/twake-web:<tag>`
+            - Example: `docker run -d -p 8080:80 --name twake-web -v /Users/linagora/twake-on-matrix/config.json:/usr/share/nginx/html/web/config.json linagora/twake-web:v2.6.5`
```
Comment (hoangdat) [inline README.md:124]: can we have a more detail example to run `twake-web` with matrix.org ?

File: lib/presentation/mixins/connect_page_mixin.dart [hunk 7]
```diff
@@ -295,6 +295,15 @@ mixin ConnectPageMixin {
...
+    html.window.history.replaceState(
+      {},
+      '',
+      '${getBaseUrlBeforeHash(html.window.location.href)}#/${route ?? 'rooms'}',
+    );
+  }
+
+  String getBaseUrlBeforeHash(String fullUrl) {
+    final fragmentIndex = fullUrl.indexOf('#/');
+    return fragmentIndex != -1 ? fullUrl.substring(0, fragmentIndex) : fullUrl;
```
Comment (hoangdat) [inline lib/presentation/mixins/connect_page_mixin.dart:307]: we should separate this fix in other PR and test carefully with some platform (saas and normal)

File: web/auth.html [hunk 8]
```diff
@@ -18,7 +18,7 @@
 
   function getOnAuthRedirectUrl() {
     let redirectUrl = window.location.href.replace('/auth.html', '/#/onAuthRedirect');
-    return redirectUrl.replace('/web', '');
+    return redirectUrl;
```
Comment (hoangdat) [inline web/auth.html:21]: we should separate this fix in other PR and test carefully with some platform (saas and normal)

File: README.md (hunk 1 above)
Comment (nqhhdev) [reply README.md:80]: Reverted

File: README.md (hunk 2 above)
Comment (nqhhdev) [reply README.md:98]: Fixed

File: README.md (hunk 3 above)
Comment (nqhhdev) [reply README.md:115]: Fixed

File: README.md (hunk 4 above)
Comment (nqhhdev) [reply README.md:117]: Fixed

File: README.md (hunk 5 above)
Comment (nqhhdev) [reply README.md:121]: Fixed

File: lib/presentation/mixins/connect_page_mixin.dart (hunk 7 above)
Comment (nqhhdev) [reply lib/presentation/mixins/connect_page_mixin.dart:307]: Move to this ticket #2067

File: web/auth.html (hunk 8 above)
Comment (nqhhdev) [reply web/auth.html:21]: Move to this ticket #2067

**Review by hoangdat — APPROVED** [review]

**Pull request merged**